step-capstone/src/__tests__/SearchForEmptyTimeslot.test.js

```diff
@@ -0,0 +1,666 @@
+
+const TWO_ITEMS_NOT_RELATE_TO_PREV_OR_NEXT_DAY = [
+  ITEM_2_0230_TO_0315,
+  ITEM_8_1700_TO_1715,
+];
+const TWO_ITEMS_ONE_FROM_PREV_ANOTHER_MIDDLE = [
+  ITEM_2300_PREV_DAY_TO_0100,
+  ITEM_5_0915_TO_1345,
+];
+const TWO_ITEMS_ONE_TO_NEXT_DAY = [
+  ITEM_5_0915_TO_1345,
+  ITEM_2300_TO_0100_NEXT_DAY,
+];
+
+const THREE_ITEMS_WITH_2_ITEMS_SAME_EDGE = [
+  ITEM_1_0115_TO_0230,
+  ITEM_2_0230_TO_0315,
+  ITEM_5_0915_TO_1345,
+];
+const THREE_ITEMS_WITH_2_SHORT_ITEMS = [
+  ITEM_8_1700_TO_1715,
+  ITEM_9_1730_TO_1745,
+  ITEM_11_2300_TO_2330,
+];
+const THREE_ITEMS_RELATE_TO_PREV_AND_NEXT_DAY = [
+  ITEM_2300_PREV_DAY_TO_0100,
+  ITEM_3_0330_TO_0500,
+  ITEM_2300_TO_0100_NEXT_DAY,
+];
+
+const FOUR_ITEMS = [
+  ITEM_1_0115_TO_0230,
+  ITEM_3_0330_TO_0500,
+  ITEM_6_1345_TO_1400,
+  ITEM_7_1430_TO_1700,
+  ITEM_8_1700_TO_1715,
+];
+const FOUR_ITEMS_RELATE_TO_PREV_AND_NEXT_DAY = [
+  ITEM_2300_PREV_DAY_TO_0100,
+  ITEM_1_0115_TO_0230,
+  ITEM_2_0230_TO_0315,
+  ITEM_2300_TO_0100_NEXT_DAY,
+];
+
+const FIVE_ITEMS = [
+  ITEM_5_0915_TO_1345,
+  ITEM_7_1430_TO_1700,
+  ITEM_9_1730_TO_1745,
+  ITEM_10_1800_TO_2245,
+  ITEM_11_2300_TO_2330,
+];
+const ONE_FULL_DAY = [ITEM_12_0000_TO_2359];
+
+// TEST handleClickedTimePoint(clickTimePoint, startOfDisplayDate, endOfDisplayDate, displayItems)
+// Assumption is that no two items in displayItems overlaps
+test("Case 0: Empty day and clickPoint at the startOfDisplayDate", () => {
+  expect(
+    handleClickedTimePoint(
+      startOfDisplayDate,
+      startOfDisplayDate,
+      endOfDisplayDate,
+      EMPTY_DAY
+    )
+  ).toStrictEqual({
+    prevTravelObject: null,
+    freeTimeSlot: { startDate: startOfDisplayDate, endDate: endOfDisplayDate },
+    nextTravelObject: null,
+  });
+});
+
+test("Case 1: Empty day and clickPoint at the endOfDisplayDate", () => {
+  expect(
+    handleClickedTimePoint(
+      endOfDisplayDate,
+      startOfDisplayDate,
+      endOfDisplayDate,
+      EMPTY_DAY
+    )
+  ).toStrictEqual({
+    prevTravelObject: null,
+    freeTimeSlot: { startDate: startOfDisplayDate, endDate: endOfDisplayDate },
+    nextTravelObject: null,
+  });
+});
+
+test("Case 2: Empty day and clickPoint at the middle of displayDate", () => {
+  expect(
+    handleClickedTimePoint(
+      TIMEPOINT_1315,
+      startOfDisplayDate,
+      endOfDisplayDate,
+      EMPTY_DAY
+    )
+  ).toStrictEqual({
+    prevTravelObject: null,
+    freeTimeSlot: { startDate: startOfDisplayDate, endDate: endOfDisplayDate },
+    nextTravelObject: null,
+  });
+});
+
+test("Case 3: ClickPoint is in travelObject", () => {
+  expect(
+    handleClickedTimePoint(
+      TIMEPOINT_1000,
+      startOfDisplayDate,
+      endOfDisplayDate,
+      ONE_ITEM_IN_THE_MIDDLE
+    )
+  ).toStrictEqual({
+    prevTravelObject: undefined,
+    freeTimeSlot: undefined,
+    nextTravelObject: undefined,
+  });
+});
+
+test("Case 4: One travelobject overflows from the previous day and clickPoint is in the rest of the day", () => {
```

Comment:
If I'm understanding this correctly, maybe this should be clarified as "clickPoint is in the part of the day after overflow event"

Reply:
Good suggestion! I'll rename the test like your suggestion.